data_layout.go

```diff
@@ -0,0 +1,421 @@
+}
+
+func getDefaultPtrAligns() map[uint64]*PointerSizeAlignment {
+	ptrAlignMap := make(map[uint64]*PointerSizeAlignment)
+	ptrAlignMap[0] = NewPointerSizeAlignment(0, 64, 64, 64, 64)
+	return ptrAlignMap
+}
+
+type IntegerSizeAlignment struct { // All sizes are in bits.
+	size               uint64
+	abiAlignment       uint64
+	preferredAlignment uint64 // optional and defaults to abi
+}
+
+func NewIntegerSizeAlignment(size, abiAl, prefAl uint64) *IntegerSizeAlignment {
+	return &IntegerSizeAlignment{size: size, abiAlignment: abiAl, preferredAlignment: prefAl}
+}
+
+func getDefaultIntAligns() map[uint64]*IntegerSizeAlignment {
+	intAlignMap := make(map[uint64]*IntegerSizeAlignment)
+	intAlignMap[1] = NewIntegerSizeAlignment(1, 8, 8)
+	intAlignMap[8] = NewIntegerSizeAlignment(8, 8, 8)
+	intAlignMap[16] = NewIntegerSizeAlignment(16, 16, 16)
+	intAlignMap[32] = NewIntegerSizeAlignment(32, 32, 32)
+	intAlignMap[64] = NewIntegerSizeAlignment(64, 32, 64)
+	return intAlignMap
+}
+
+type VectorSizeAlignment struct { // All sizes are in bits.
+	size               uint64
+	abiAlignment       uint64
+	preferredAlignment uint64 // optional and defaults to abi
+}
+
+func NewVectorSizeAlignment(size, abiAl, prefAl uint64) *VectorSizeAlignment {
+	return &VectorSizeAlignment{size: size, abiAlignment: abiAl, preferredAlignment: prefAl}
+}
+
+func getDefaultVectorAligns() map[uint64]*VectorSizeAlignment {
+	vectorAlignMap := make(map[uint64]*VectorSizeAlignment)
+	vectorAlignMap[64] = NewVectorSizeAlignment(64, 64, 64)
+	vectorAlignMap[128] = NewVectorSizeAlignment(128, 128, 128)
+	return vectorAlignMap
+}
+
+type FloatingPointSizeAlignment struct { // All sizes are in bits.
+	size               uint64
+	abiAlignment       uint64
+	preferredAlignment uint64 // optional and defaults to abi
+}
+
+func NewFloatingPointSizeAlignment(size, abiAl, prefAl uint64) *FloatingPointSizeAlignment {
+	return &FloatingPointSizeAlignment{size: size, abiAlignment: abiAl, preferredAlignment: prefAl}
+}
+
+func getDefaultFloatAligns() map[uint64]*FloatingPointSizeAlignment {
+	floatAlignMap := make(map[uint64]*FloatingPointSizeAlignment)
+	floatAlignMap[16] = NewFloatingPointSizeAlignment(16, 16, 16)
+	floatAlignMap[32] = NewFloatingPointSizeAlignment(32, 32, 32)
+	floatAlignMap[64] = NewFloatingPointSizeAlignment(64, 64, 64)
+	floatAlignMap[128] = NewFloatingPointSizeAlignment(128, 128, 128)
+	return floatAlignMap
+}
+
+type AggregateAlignment struct {
+	abiAlignment       uint64
+	preferredAlignment uint64 // optional and defaults to abi
+}
+
+func NewAggregateAlignment(abiAl, prefAl uint64) *AggregateAlignment {
+	return &AggregateAlignment{abiAlignment: abiAl, preferredAlignment: prefAl}
+}
+
+type FunctionPointerAlignment struct {
+	isIndependant bool
+	abiAlignment  uint64
+}
+
+func NewFunctionPointerAlignment(isInd bool, abiAl uint64) *FunctionPointerAlignment {
+	return &FunctionPointerAlignment{isIndependant: isInd, abiAlignment: abiAl}
+}
+
+type ManglingStyle string
```

Comment:
Consider moving `ManglingStyle` to the `ir/enum` package. We may consider making an integer enum with `String` methods generated using `stringer`. See https://github.com/llir/llvm/blob/9e62c17a9f75d127c4010b0d242912413c9d73cc/ir/enum/enum.go#L42 for an example.

Reply:
I think we should do this later; else a PR depends on another PR will be hard to manage 

And we can do that easier since we are owner